nautobot/dcim/factory.py

```diff
@@ -99,6 +99,115 @@ def get_rack_reservation_units(obj):
     return [unit for unit in available_units if unit not in unavailable_units][:1]
 
 
+class DeviceFactory(PrimaryModelFactory):
+    class Meta:
+        model = Device
+        exclude = (
+            "has_tenant",
+            "has_platform",
+            "has_serial",
+            "has_asset_tag",
+            "has_device_redundancy_group",
+            "has_comments",
+        )
+
+    device_type = random_instance(DeviceType, allow_null=False)
+    status = random_instance(
+        lambda: Status.objects.get_for_model(Device),
+        allow_null=False,
+    )
+    role = random_instance(
+        lambda: Role.objects.get_for_model(Device),
+        allow_null=False,
+    )
+    location = random_instance(
+        lambda: Location.objects.filter(location_type__content_types=ContentType.objects.get_for_model(Device)),
+        allow_null=False,
+    )
+    name = factory.LazyAttributeSequence(lambda o, n: f"{o.device_type.model} Device - {o.location} - {n + 1}")
+
+    has_tenant = NautobotBoolIterator()
+    tenant = factory.Maybe("has_tenant", random_instance(Tenant))
+    has_platform = NautobotBoolIterator()
+    platform = factory.Maybe(
+        "has_platform",
+        factory.LazyAttribute(lambda o: Platform.objects.filter(manufacturer=o.device_type.manufacturer).first()),
```

Comment:
```
factory.LazyAttribute(lambda o: random_instance(Platform.objects.filter(manufacturer=o.device_type.manufacturer))),
```
I wanted to use random_instance() but it would not let me. This is working for now.
This is the error I got:
```
ValueError: Cannot assign "<factory.declarations.LazyFunction object at 0x7fc76e256a40>": "Device.platform" must be a "Platform" instance.
```

Reply:
If you want a random Platform from the filtered queryset, you can use `.order_by("?").first()` - it's slow at scale but might be fine here.

Reply:
I think you need to do it like this:

```py
def get_random_platform_for_manufacturer(manufacturer):
    qs = Platform.objects.filter(manufacturer=manufacturer)
    return factory.random.randgen.choice(qs) if qs.exists() else None

...

    platform = factory.Maybe(
        "has_platform",
        factory.LazyAttribute(lambda o: get_random_platform_for_manufacturer(o.device_type.manufacturer)),
        None,
    )
```